### PR TITLE
Feat/textbox view options

### LINKED ISSUE #384: Textbox view options
### Expected behavior

* [ ] Box is always full browser height

screen shot 2018-06-06 at 14 48 04
screen shot 2018-06-06 at 14 48 11

* [ ] Fixed number of rows
* [ ] Fit to content

### Actual behavior

Only resizable by dragging bottom right corner

### PR DESCRIPTION
**Pull request by danwdart**
Yes, I know `this.base.style` is TOTALLY the wrong way to do it, but what am I going to do, re-render self? Is that the correct way of doing it in this case?

Fixes #384

### CODE REVIEW

File: frontend/components/TextInput/TextInput.jsx [hunk 1]
```diff
@@ -104,21 +126,35 @@ export default class TextInput extends Component {
...
     } = this.props
 
     let inputStyle = new Style(styles, 'input')
 
     inputStyle.addIf('input-in-label', inLabel)
+    inputStyle.add(resizable ? `resizable` : `not-resizable`)
+    if (heightType === `full`) {
+      inputStyle.add(`full-height`)
+    }
+    else if (heightType === `content`) {
```
Comment (eduardoboucas) [inline frontend/components/TextInput/TextInput.jsx]: `else` in the same line as `if`, please 😇

File: frontend/components/TextInput/TextInput.jsx [hunk 2]
```diff
@@ -104,21 +126,35 @@ export default class TextInput extends Component {
...
+      resizable,
+      rows,
       type,
       value
     } = this.props
 
     let inputStyle = new Style(styles, 'input')
 
     inputStyle.addIf('input-in-label', inLabel)
+    inputStyle.add(resizable ? `resizable` : `not-resizable`)
```
Comment (eduardoboucas) [inline frontend/components/TextInput/TextInput.jsx]: Any reason to use template literals in place of regular strings?

Comment (danwdart) [reply frontend/components/TextInput/TextInput.jsx]: I'm used to it. No other reason. It's also less work to add templates if I need to, so I've just started to default to it. I'm all for not doing so if that's the rules / we're using `standard` / etc. Let's eslint the lot, really.

File: frontend/components/TextInput/TextInput.jsx (hunk 1 above)
Comment (danwdart) [reply frontend/components/TextInput/TextInput.jsx]: Oh you mean `} else  if (` ? Of course, missed that one.

**Comment by jimlambie** [comment]
http://publish-feat-textbox-view-options.eb.dev.dadi.technology/articles2/new

File: frontend/components/TextInput/TextInput.css [hunk 3]
```diff
@@ -9,4 +9,20 @@
...
+.resizable {
+  resize: both;
+}
+
+.not-resizable {
+  resize: none;
+}
+
+.full-height {
+  height: 100vh;
```
Comment (abovedave) [inline frontend/components/TextInput/TextInput.css]: Maybe this should account for the save bar e.g., ` calc(100vh - 135px)`?

Comment (danwdart) [reply frontend/components/TextInput/TextInput.css]: Sounds good, I'll add it

**Comment by danwdart** [comment]
Definitely needs a few more tests (I'm adding)

**Pull request merged**